src/main/webapp/script.js

```diff
@@ -23,24 +23,28 @@ google.setOnLoadCallback(getChart)
 
 async function getYouTubeComments() { 
   const urlInput = document.getElementById('link-input');
-  const url = cleanseUrl(urlInput.value);
+  const url = extractYouTubeUrl(urlInput.value);
   const response = await fetch("/YouTubeComments?url="+url);
   const comments = await response.json();
   return comments;
 }
 
 /*
- * Extracts video id from full url
- */ 
-function cleanseUrl(url) {
-  // Split web address from parameters, extract first parameter
-  // TODO: Add checks to make this work if video is not first parameter.
-  var videoId = url.split("?");
-  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
-
-  // If param name present, remove it to isolate video Id.
-  videoId = videoId.replace("v=", "");
-  
+ * Extracts video ID from the full URL
+ * Method derived from: 
+ * https://stackoverflow.com/questions/28735459/how-to-validate-youtube-url-in-client-side-in-text-box
+ */
+function extractYouTubeUrl(url) {
```

Comment:
Can we test this method?

Reply:
Is there a way for Maven to test javascript functions?

Reply:
Anything is always possible, and there are many ways to do it.

There are two primary steps that you need to focus on:
1. Find a javascript library to use for writing unit tests (Mocha, Jest, Jasmine, ... etc.)
2. Add a maven plugin to your `pom.xml` that executes these unit tests when packaging your project.

Here's an [article](https://dimitr.im/jasmine-tests-maven) that gives an example of testing javascript using maven.

One other resource you can take advantage of is searching other google intern repositories to see how they tackled similar problems. Github has a nice built-in search to service this need:
[This example searches for java servlet tests](https://github.com/search?q=org%3Agoogleinterns+language%3Ajava+filename%3ATest.java+filename%3Aservlet&type=Code)